src/cmap/worker.rs

```diff
@@ -483,6 +508,28 @@ impl ConnectionPoolWorker {
         }
     }
 
+    fn store_pinned(&mut self, mut conn: Connection) {
+        if self.pending_unpins.remove(&conn.id) {
+            conn.unpin();
+            self.check_in(conn);
+        } else {
+            self.pinned_connections.insert(conn.id, conn);
```

Comment:
We're going to need to check if a connection was dropped while it was reading/writing and, if so, check it back in (which will close it) rather than keeping it pinned. Otherwise the next time it's used, it may read data from a previous operation. See line 504 for where this check happens for normal check ins.

Reply:
Done.